frontend/apps/reader/readerui.lua

```diff
@@ -356,12 +356,11 @@ function ReaderUI:showReader(file)
     end)
 end
 
-local running_instance = nil
 function ReaderUI:doShowReader(file)
     dbg("opening file", file)
     -- keep only one instance running
-    if running_instance then
-        running_instance:onClose()
+    if self.instance then
```

Comment:
This needs to be a global variable inside the module. If each ReaderUI instance has its own value, the onClose method will only update its own instance key and doShowReader won't be able to pick it up.